### PR TITLE
Implemented support for tar.gz and tar.bz archives

### PR DESCRIPTION
**Pull request by tglman**
Hi,

here it is the pull request, it's not yet final, code should be reformatted and it may be needed a double check on file extensions included in the stripping.

Anyway a quick review and feedback is welcome.

Regards

### CODE REVIEW

**Comment by Zlika** [comment]
Thank you for this PR.

I've added to the repo an Eclipse formatter and also Checkstyle rules to enforce a consistent coding style. The main requirements are:
* Encoding: UTF-8
* Brackets: At the beginning of a line (Maven convention)
* Line size: 120 char max

Now with the use of the maven-checkstyle-plugin, if the build passes then the coding style is ok. You can use the Eclipse checkstyle plugin in your IDE too.

Concerning your code:
* It would be nice to also add an integration test to check at least that the timestamps have been removed from the tarballs. You can take the integration test https://github.com/Zlika/reproducible-build-maven-plugin/tree/master/src/it/same-jar as an example. There are two main sources of non reproducible data in archives: the file order (which may depend on the order of the files in the filesystem of the user) and timestamps. It's difficult to test the first one, but the second one is easier, you just have to build twice and check that the hashes are equals.
* You process tar.gz and tar.bz the same way as zip files (in particular, you try to strip MANIFEST.MF, pom.properties...) but this is not relevant here. It is done for ZIP files because JAR file are ZIP files. However, tar.gz/bz files are more likely produced by the maven-assembly-plugin, so there is no need to check for these files here. Also, as the maven-assembly-plugin "single" goal generally binds to the "package" phase, it can include JAR files that were not processed by the reproducible-build-maven-plugin yet. This can be a problem, however I don't know how to do, other than asking users to bind the maven-assembly-plugin to a latter phase (I don't want to recursively strip all the JAR files in an archive, because some may be dependencies and it would not make sense to strip them, and it may even not be allowed depending on their license).
* I don't see the need for the AbstractStripper class. Maybe it's just that I'm not fond of abstract classes, but it seems to me that it adds complexity for a very minimal benefit.

File: src/main/java/io/github/zlika/reproducible/TarStripper.java [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+				.collect(Collectors.toList());
+		return sortedNames;
+	}
+
+	private TarArchiveEntry filterTarEntry(TarArchiveEntry entry) {
+		// Set times
+		entry.setModTime(0);
+		// Remove GID and UID that are not deterministic, to evaluate though
+		entry.setGroupId(1000);
+		entry.setUserId(1000);
```
Comment (Zlika) [inline src/main/java/io/github/zlika/reproducible/TarStripper.java]: That's a good idea, however you should check what happens if you try to untar the file on a system that does not know this ID (or that knows it but is effected to another user than the one who untars the file).
Also it seems that there are other non reproducible fields in a TAR entry, e.g. userName and groupName (cf. https://svn.apache.org/repos/asf/ant/core/trunk/src/main/org/apache/tools/tar/TarEntry.java).

**Comment by tglman** [comment]
hi, 

Thank you for the review, i'll do some changes and a bit more tests and come back to you.

Regards

**Comment by tglman** [comment]
Hi,

I removed no need jar related stripper from the tar stripper, the abstract class that has no meaning anymore, and applied the code style, I still missing the integration test though, it will come soon.

Regards

**Comment by tglman** [comment]
hi @Zlika ,

I'm implementing the Integration tests, how do you run the Integration them ?

Regards

**Comment by Zlika** [comment]
Hi, 
mvn clean install - Prun-its

**Comment by unicolet** [comment]
@Zlika since we need this feature my [company](http://www.proemion.com) could allow me to work on getting this PR finalized. I'd pick it up from where it is now and work from there: wdyt, would that work for you?

/cc @tglman

**Comment by Zlika** [comment]
Hi!
Feel free to work on this feature.
An important difference between zip and tar on a reproducibility point of view is that tar files include the uid/guid of the files, so you should find a way to fix that in a reproducible and portable way.
I'm waiting forward to merging your PR :-)

**Comment by Zlika** [comment]
Replaced by #14.

**Pull request closed without merging**